repo/content/content_manager_indexes.go

```diff
@@ -87,48 +102,52 @@ func (bm *Manager) compactAndDeleteIndexBlobs(ctx context.Context, indexBlobs []
 		return nil
 	}
 
-	formatLog(ctx).Debugf("compacting %v contents", len(indexBlobs))
+	formatLog(ctx).Debugf("compacting %v index blobs", len(indexBlobs))
 
-	t0 := time.Now() // allow:no-inject-time
 	bld := make(packIndexBuilder)
 
+	var inputs, outputs []blob.Metadata
+
 	for _, indexBlob := range indexBlobs {
 		if err := bm.addIndexBlobsToBuilder(ctx, bld, indexBlob, opt); err != nil {
-			return err
+			return errors.Wrap(err, "error adding index to builder")
 		}
+
+		inputs = append(inputs, indexBlob.Metadata)
 	}
 
 	var buf bytes.Buffer
 	if err := bld.Build(&buf); err != nil {
 		return errors.Wrap(err, "unable to build an index")
 	}
 
-	compactedIndexBlob, err := bm.writePackIndexesNew(ctx, buf.Bytes())
+	compactedIndexBlob, err := bm.indexBlobManager.writeIndexBlob(ctx, buf.Bytes())
 	if err != nil {
 		return errors.Wrap(err, "unable to write compacted indexes")
 	}
 
-	formatLog(ctx).Debugf("wrote compacted index (%v bytes) in %v", compactedIndexBlob, time.Since(t0)) // allow:no-inject-time
-
+	// compaction wrote index blob that's the same as one of the sources
+	// it must be a no-op.
 	for _, indexBlob := range indexBlobs {
-		if indexBlob.BlobID == compactedIndexBlob {
-			continue
+		if indexBlob.BlobID == compactedIndexBlob.BlobID {
+			formatLog(ctx).Debugf("compaction was a no-op")
+			return nil
 		}
+	}
```

Comment:
1. Can this case occur only when there is just one indexBlob in indexBlobs to compact?
2. Won't PutBlob return an error when indexBlobManager.writeIndexBlob tries to write a blob with the same ID?